.ruff.toml

```diff
@@ -21,5 +21,90 @@
 # FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
 # IN THE SOFTWARE.
 
+target-version = "py310"
 line-length = 80
 fix = true
+
+[lint]
+select = [
+    "A", "ANN", "ARG",
+    "B", "BLE",
+    "C", "C4",
+    "COM",
+    "D", "DTZ",
+    "E", "ERA", "EXE",
+    "F", "FBT",
+    "G",
+    "I", "ICN", "INP", "ISC",
+    "N",
+    "PGH", "PIE", "PL", "PLC", "PLE", "PLR", "PLW", "PT", "PYI",
+    "Q",
+    "RUF", "RSE", "RET",
+    "S", "SIM", "SLF",
+    "T", "T10", "T20", "TCH", "TID", "TRY",
+    "UP",
+    "W",
+    "YTT",
+]
+ignore = [
+    "A001",  # Variable is shadowing a Python builtin
+    "ANN204",  # Missing return type annotation for special method __str__
+    "ANN401",  # Dynamically typed expressions (typing.Any) are disallowed
+    "ARG005",  # Unused lambda argument
+    "C901",  # Too complex
+    "COM812",  # Conflicts with formatting
+    "D100",  # Missing docstring in public module
+    "D102",  # Missing docstring in public method
+    "D104",  # Missing docstring in public package
+    "D105",  # Missing docstring in magic method
+    "D107",  # Missing docstring in `__init__` method
+    "D417",  # Missing argument description in the docstring
+    "E501",  # Line too long
+    "ERA001",  # Commented out code
+    "FBT001",  # Boolean positional parameter in function signature
+    "FBT003",  # Boolean positional value in function call
+    "G004",  # Logging statement uses f-string
+    "PLR0911",  # Too many return statements
+    "PLR0912",  # Too many branches
+    "PLR0913",  # Too many arguments to function call
+    "PLR0915",  # Too many statements
+    "RUF003",  # Ambiguous emdash/endash
+    "SIM108",  # Consider ternary operator
+    "SLF001", # Private member accessed
+    "TRY003",  # Avoid specifying long messages outside the exception class
+]
+
+[lint.per-file-ignores]
+"python/**.py" = [
+    "S110",  # Consider logging exceptions
+]
+"python/zensical/main.py" = [
+    "T201",  # Print statements
+]
+"python/**.pyi" = [
+    "PYI021",  # Docstrings in stubs
+]
+"scripts/*.py" = [
+    "PLR2004",  # Magic value used in comparison
+    "S603",  # Possible untrusted input
+    "S605",  # Starting process with shell=True
+    "S607",  # Subprocess with partial executable path
+    "T201",  # Print statements
+]
+
+[lint.flake8-quotes]
+docstring-quotes = "double"
+
+[lint.flake8-tidy-imports]
+ban-relative-imports = "all"
```

Comment:
I prefer absolute imports, but I don't mind if we keep relative ones, no strong opinions.